kodeagent.py

```
@@ -1836,6 +1838,146 @@ async def _act(self) -> AsyncIterator[AgentResponse]:
+
+        # Filter the agent's tools based on the LLM's response
+        self.tools = [
+            tool_func for tool_func in self.original_tools if tool_func.name in relevant_tool_names
+        ]
+        self.tool_names = set(relevant_tool_names)
+        self.tool_name_to_func = {name: func for name, func in zip(self.tool_names, self.tools)}
+
+        # Update tools_source_code as well, this is important for CodeActAgent
+        self.tools_source_code = 'from typing import *\n\nimport kutils as ku\n\n'
+        for t in self.tools:  # Use the filtered list of tools
+            self.tools_source_code += inspect.getsource(t).strip('@tool') + '\n'
```

Ruff (0.11.9)

Using `.strip()` with multi-character strings is misleading

(B005)